::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2017 Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Guillermo Herrero <guillermo.herrero@enea.com>
Date: Wed, 4 Oct 2017 18:26:16 +0200
Subject: [PATCH] classes: baremetal: AArch64: NIC adaptation 

AArch64 NIC adaptation for dual NIC boards with traffic separation
 -kvm nodes: added second trunk_nic
 -compute nodes: removed third external_nic
 
Change-Id: I42c3511e3d0820213206574fab30dbd3bd84b1c0
Signed-off-by: Guillermo Herrero <guillermo.herrero@enea.com>
---
 .../cluster/baremetal-mcp-ocata-odl-ha/infra/kvm.yml      | 12 +++++++++---
 .../baremetal-mcp-ocata-odl-ha/openstack/compute.yml      | 15 ++++-----------
 .../cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/kvm.yml | 14 ++++++++++----
 .../cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml      | 12 +++++++++---
 .../baremetal-mcp-ocata-ovs-ha/openstack/compute.yml      | 13 +++----------
 5 files changed, 35 insertions(+), 31 deletions(-)

diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/infra/kvm.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/infra/kvm.yml
index c13c475..bbaaebb 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/infra/kvm.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/infra/kvm.yml
@@ -26,6 +26,7 @@ parameters:
     keepalived_vip_interface: br-ctl
     keepalived_vip_virtual_router_id: 69
     deploy_nic: ${_param:openstack_primary_nic}
+    trunk_nic: ${_param:openstack_secondary_nic}
   salt:
     control:
       size:  # RAM 4096,8192,16384,32768,65536
@@ -148,6 +149,11 @@ parameters:
           netmask: 255.255.255.0
           name: ${_param:deploy_nic}
           noifupdown: true
+        trunk:
+          enabled: true
+          type: eth
+          proto: manual
+          name: ${_param:trunk_nic}
         br-mgmt:
           enabled: true
           proto: dhcp
@@ -160,9 +166,9 @@ parameters:
           enabled: true
           proto: manual
           type: vlan
-          name: ${_param:deploy_nic}.${_param:opnfv_net_mgmt_vlan}
+          name: ${_param:trunk_nic}.${_param:opnfv_net_mgmt_vlan}
           use_interfaces:
-            - ${_param:deploy_nic}
+            - ${_param:trunk_nic}
         br-ctl:
           enabled: true
           type: bridge
@@ -170,4 +176,4 @@ parameters:
           address: ${_param:single_address}
           netmask: 255.255.255.0
           use_interfaces:
-            - ${_param:deploy_nic}.${_param:opnfv_net_mgmt_vlan}
+            - ${_param:trunk_nic}.${_param:opnfv_net_mgmt_vlan}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/openstack/compute.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/openstack/compute.yml
index 841d2f2..f79b5ea 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/openstack/compute.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/openstack/compute.yml
@@ -26,7 +26,6 @@ parameters:
     nova_vncproxy_url: https://${_param:cluster_public_host}:6080
     mgmt_nic: ${_param:openstack_primary_nic}
     tenant_nic: ${_param:openstack_secondary_nic}
-    external_nic: ${_param:openstack_tertiary_nic}
     interface_mtu: 1500
     keepalived_vip_interface: br-ctl
     keepalived_vip_virtual_router_id: 69
@@ -77,12 +76,6 @@ parameters:
           type: eth
           proto: manual
           name: ${_param:tenant_nic}
-        external_nic:
-          enabled: true
-          name: ${_param:external_nic}
-          mtu: ${_param:interface_mtu}
-          proto: manual
-          type: eth
         br-mesh:
           enabled: true
           type: bridge
@@ -95,9 +88,9 @@ parameters:
           enabled: true
           proto: manual
           type: vlan
-          name: ${_param:mgmt_nic}.${_param:opnfv_net_mgmt_vlan}
+          name: ${_param:tenant_nic}.${_param:opnfv_net_mgmt_vlan}
           use_interfaces:
-            - ${_param:mgmt_nic}
+            - ${_param:tenant_nic}
         tenant-vlan:
           enabled: true
           proto: manual
@@ -112,7 +105,7 @@ parameters:
           address: ${_param:single_address}
           netmask: 255.255.255.0
           use_interfaces:
-            - ${_param:mgmt_nic}.${_param:opnfv_net_mgmt_vlan}
+            - ${_param:tenant_nic}.${_param:opnfv_net_mgmt_vlan}
         br-floating:
           enabled: true
           type: ovs_bridge
@@ -129,6 +122,6 @@ parameters:
           address: ${_param:external_address}
           netmask: 255.255.255.0
           use_interfaces:
-            - ${_param:external_nic}
+            - ${_param:tenant_nic}
           use_ovs_ports:
             - float-to-ex
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/kvm.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/kvm.yml
index fedae85..50db055 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/kvm.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/kvm.yml
@@ -25,7 +25,8 @@ parameters:
     cluster_node03_address: ${_param:infra_kvm_node03_address}
     keepalived_vip_interface: br-ctl
     keepalived_vip_virtual_router_id: 69
-    deploy_nic: ${_param:openstack_primary_nic}
+    deploy_nic: ${_param:openstack_primary_nic
+    trunk_nic: ${_param:openstack_secondary_nic}
   salt:
     control:
       size:  # RAM 4096,8192,16384,32768,65536
@@ -137,6 +138,11 @@ parameters:
           netmask: 255.255.255.0
           name: ${_param:deploy_nic}
           noifupdown: true
+        trunk:
+          enabled: true
+          type: eth
+          proto: manual
+          name: ${_param:trunk_nic}
         br-mgmt:
           enabled: true
           proto: dhcp
@@ -149,9 +155,9 @@ parameters:
           enabled: true
           proto: manual
           type: vlan
-          name: ${_param:deploy_nic}.${_param:opnfv_net_mgmt_vlan}
+          name: ${_param:trunk_nic}.${_param:opnfv_net_mgmt_vlan}
           use_interfaces:
-            - ${_param:deploy_nic}
+            - ${_param:trunk_nic}
         br-ctl:
           enabled: true
           type: bridge
@@ -159,4 +165,4 @@ parameters:
           address: ${_param:single_address}
           netmask: 255.255.255.0
           use_interfaces:
-            - ${_param:deploy_nic}.${_param:opnfv_net_mgmt_vlan}
+            - ${_param:trunk_nic}.${_param:opnfv_net_mgmt_vlan}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml
index 63fdd7a..90496dc 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml
@@ -26,6 +26,7 @@ parameters:
     keepalived_vip_interface: br-ctl
     keepalived_vip_virtual_router_id: 69
     deploy_nic: ${_param:openstack_primary_nic}
+    trunk_nic: ${_param:openstack_secondary_nic}
   salt:
     control:
       size:  # RAM 4096,8192,16384,32768,65536
@@ -138,6 +139,11 @@ parameters:
           netmask: 255.255.255.0
           name: ${_param:deploy_nic}
           noifupdown: true
+        trunk:
+          enabled: true
+          type: eth
+          proto: manual
+          name: ${_param:trunk_nic}
         br-mgmt:
           enabled: true
           proto: dhcp
@@ -150,9 +156,9 @@ parameters:
           enabled: true
           proto: manual
           type: vlan
-          name: ${_param:deploy_nic}.${_param:opnfv_net_mgmt_vlan}
+          name: ${_param:trunk_nic}.${_param:opnfv_net_mgmt_vlan}
           use_interfaces:
-            - ${_param:deploy_nic}
+            - ${_param:trunk_nic}
         br-ctl:
           enabled: true
           type: bridge
@@ -160,4 +166,4 @@ parameters:
           address: ${_param:single_address}
           netmask: 255.255.255.0
           use_interfaces:
-            - ${_param:deploy_nic}.${_param:opnfv_net_mgmt_vlan}
+            - ${_param:trunk_nic}.${_param:opnfv_net_mgmt_vlan}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/compute.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/compute.yml
index 422f65d..38c9e7f 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/compute.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/compute.yml
@@ -25,7 +25,6 @@ parameters:
     nova_vncproxy_url: https://${_param:cluster_public_host}:6080
     mgmt_nic: ${_param:openstack_primary_nic}
     tenant_nic: ${_param:openstack_secondary_nic}
-    external_nic: ${_param:openstack_tertiary_nic}
     linux_system_codename: xenial
     interface_mtu: 1500
     keepalived_vip_interface: br-ctl
@@ -60,12 +59,6 @@ parameters:
           type: eth
           proto: manual
           name: ${_param:tenant_nic}
-        external_nic:
-          enabled: true
-          name: ${_param:external_nic}
-          mtu: ${_param:interface_mtu}
-          proto: manual
-          type: eth
         br-mesh:
           enabled: true
           type: bridge
@@ -78,9 +71,9 @@ parameters:
           enabled: true
           proto: manual
           type: vlan
-          name: ${_param:mgmt_nic}.${_param:opnfv_net_mgmt_vlan}
+          name: ${_param:tenant_nic}.${_param:opnfv_net_mgmt_vlan}
           use_interfaces:
-            - ${_param:mgmt_nic}
+            - ${_param:tenant_nic}
         tenant-vlan:
           enabled: true
           proto: manual
@@ -112,6 +105,6 @@ parameters:
           address: ${_param:external_address}
           netmask: 255.255.255.0
           use_interfaces:
-            - ${_param:external_nic}
+            - ${_param:tenant_nic}
           use_ovs_ports:
             - float-to-ex
